tests/behat/features/bootstrap/Page/SearchPage.php

```diff
@@ -0,0 +1,1391 @@
+    $max = 300;
+    $found = $this->find('css', $this->elements['autocomplete']);
+    while (--$max > 0 && !$found) {
+      usleep(100);
+      $found = $this->find('css', $this->elements['autocomplete']);
+    }
+
+    // Report error if we ran out of time.
+    if (!$found) {
+      return "Openscan did not show any suggestions. ";
+    }
+
+    $found = $this->getElement('autocomplete');
+
+    // It also takes a bit for the page to get the dynamics of the suggestions done. So we wait again.
+    $max = 300;
+    $cnt = 0;
+    while (--$max > 0 && !$found->findAll("css", $this->elements['autocomplete-list'])) {
+      usleep(100);
+      // Refresh the search.
+      $found = $this->getElement('autocomplete');
+    }
+
+    // Now we list the suggestions given.
+    foreach ($found->findAll("css", $this->elements['autocomplete-list']) as $suggestion) {
+      $this->logMsg(true, $suggestion->getText() . "\n");
+      $cnt++;
+    }
+    if ($cnt == 0) {
+      return "No suggestions were found.";
+    }
+    // All we can do is list the number for convenience. It's in the configuration how many there should be.
+    $this->logMsg(true, "In total " . $cnt . " suggestions were shown. Check configurationen.");
+    return "";
+  }
+
+  /**
+   * Tries to find the number of search results shown as total found posts on the search page.
+   *
+   * @return int
+   *    The number of search results.
+   */
+  public function getShownSizeOfSearchResult() {
+    // todo: use getElement here.
+    $found = $this->find('css', '.pane-content .count');
+    if (!$found) {
+      $this->logMsg(true, "Couldn't find count of results on page.");
+      return -1;
+    }
+
+    // GetText() will here be of the form "(nnn resultater)", where nnn is what we want.
+    // So we will split the string on the space, then substr the '(' away.
+    $resArr = explode(' ', $found->getText());
+    if (count($resArr) == 0) {
+      return "Couldn't interpret result as '(x resultater)'. All I found was this: " . $found->getText();
+    }
+    $expectCount = substr($resArr[0], 1);
+
+    return $expectCount;
+  }
+
+  /**
+   * Returns the verbose setting for search result
+   *
+   * @return string
+   *   NonEmpty if failure.
+   */
+  public function getVerboseSearchResult() {
+    return $this->verboseSearchResults;
+  }
+
+
+  /**
+   * GoToPage.
+   *
+   * @param int $toPage
+   *    Pagenumber to go to.
+   *
+   * @return string
+   *    Nonempty in case of failure.
+   *
+   * @throws Exception
+   *    In case of error.
+   */
+  public function goToPage($toPage) {
+    // This is counter intuitive, but page 2 will have parameter "page=1", so we start by subtracting 1.
+    $toPage = $toPage - 1;
+    $paginations = $this->findAll('xpath', $this->elements['pager-links']);
+    // Initialise link - the index into the array.
+    $link = -2;
+
+    for ($i = 0; $i < count($paginations); $i++) {
```

Comment:
count($paginations) should be done outside loop declaration for better performance

Reply:
Fixed.